plugins/chat/assets/javascripts/discourse/routes/chat.js

```diff
@@ -65,8 +65,10 @@ export default class ChatRoute extends DiscourseRoute {
   }
 
   deactivate(transition) {
-    const url = this.router.urlFor(transition.from.name);
-    this.chatStateManager.storeChatURL(url);
+    if (transition) {
```

Comment:
can you expand on when does this case happen please?

Reply:
It happens when `deactivate` is called from the route's `willDestroy` method, in that case, there is no transition. So the transition argument is nullable:
https://github.com/emberjs/ember.js/blob/06735dbba898457afc9acbc551ae389d06b02306/packages/%40ember/routing/route.ts#L857

This place fails in acceptance tests, because at the end of an acceptance test, Ember just destroys the app and the route.

As far as I know this will never happen in a real app, for example see https://discuss.emberjs.com/t/are-routes-destroyed-while-transitioning/17644.